keras/layers/recurrent.py

```diff
@@ -217,7 +218,15 @@ def call(self, x, mask=None):
                                 'argument, including the time axis. '
                                 'Found input shape at layer ' + self.name +
                                 ': ' + str(input_shape))
-        if self.stateful:
+        if self.initial_states is not None:
+            if self.stateful:
+                raise Exception('Keras does not currently support stateful Recurrent '
+                                'layers with custom initial states.')
+            if type(self.initial_states) in (list, tuple):
+                initial_states = self.initial_states
+            else:
+                initial_states = [self.initial_states]
```

Comment:
What @braingineer said is correct but it's not a Numpy array. In the application where I've been using this, I use the output of another layer and it works fine. @fchollet, not sure what you mean by it being a symbolic variable.